src/GeneratedMapper/Builders/MappingBuilder.cs

```diff
@@ -14,10 +14,15 @@ namespace GeneratedMapper.Builders
     internal sealed class MappingBuilder : BuilderBase
     {
         private readonly List<PropertyMappingBuilder> _propertyMappingBuilders;
-
+        private readonly ParameterInformation[] _mapArgumentParmeters;
+        private readonly ParameterInformation[] _afterMapParameters;
+        
         public MappingBuilder(MappingInformation information) : base(information)
         {
             _propertyMappingBuilders = information.Mappings.Select(mapping => new PropertyMappingBuilder(mapping, SourceInstanceName)).ToList();
+
+            _mapArgumentParmeters = _propertyMappingBuilders.SelectMany(x => x.MapArgumentsRequired()).ToArray();
```

Comment:
These two private variables are only used together in a `Concat`. Why not concat here?

Reply:
I was keeping them separate as part of refactoring, and thought I might need them in a case like expression mapping.  Then after looking over resolvers and extension methods I realized that Expression mapping follows a completely different code path for determining the parameters.

I realized they couldn't be synced in any meaningful fashion currently, because Expressions don't use async expressions, but I left it like that just in case to refactor into the base class somehow at a later date.  I don't know if that's worth the effort or not at this point.